### DISCUSSION TITLE
ReasonCode (BT-98) for SpecifiedTradeAllowanceCharge (BG-20) on document level

### DISCUSSION THREAD

**Discussion by jr-hoh** [stephanstapel/ZUGFeRD-csharp#406] [Ideas]
Hi together,

I couldn't find anything in the code regarding the ReasonCode (BT-98) for allowances and charges on document level.
There is only the possibility to add the Reason (BT-97) in text form. Is there a special reason for that?

Cheers

**Comment by stephanstapel** [comment]
Hi @HiCodingOtter ,
It would certainly be a good idea to add it since it is part of the standard since ZUGFeRD 1.x.
If you like, implement it and send a pull request. I'm happy to review it!

**Reply by jr-hoh** [reply]
When I find the time, I will do it and will be happy to support the project :)

**Comment by tobitege** [comment]
Put up a PR for this: #575

**Comment by tobitege** [comment]
@jr-hoh The PR has been merged and you can now pass the `ReasonCode` via the `AddTradeAllowanceCharge` methods (if you update your sources from `master` branch).
